## DatasetCollection-JBEI_plasmid_test
### Input
```yaml
datasets:
- bibliographicCitation: https://public-registry.jbei.org/folders/171
  brc: JBEI
  creator:
  - affiliation: Joint BioEnergy Institute, Lawrence Berkeley National Laboratory
      Berkeley CA 94720
    email: ''
    name: Robert Dahl
    primaryContact: true
  datasetName: JBEI-3807
  date: '2011-07-08'
  description: 'fadB promoter driving RFP in E. coli Selection Markers: Kanamycin
    Host: E. coli DH1'
  identifier: JPUB_005436
  keywords: []
  plasmid_features:
  - backbone: ''
    host:
      NCBITaxID: 536056
      scientificName: Escherichia coli DH1
    ori: ''
    promoters:
    - fadB
    replicates_in:
    - NCBITaxID: 536056
      scientificName: Escherichia coli DH1
    selection_markers:
    - kan
  repository: ICE
  title: JBEI-3807

```
## DatasetCollection-JBEI_examples_negative
### Input
```yaml
datasets:
- bibliographicCitation: https://dx.doi.org/10.1038/s41467-024-46314-8
  brc: JBEI
  creator:
  - affiliation: Joint BioEnergy Institute, Lawrence Berkeley National Laboratory
      Berkeley CA 94720
    creatorName: Christopher J. Petzold
    email: cjpetzold@lbl.gov
    primaryContact: true
  date: '2024-03-22'
  identifier: PXD043152
  keywords:
  - Bioengineering
  - Edible mycelium
  - nutrition
  - synthetic biology
  relatedItem:
    relatedItemIdentifier: https://www.nature.com/articles/s41467-024-46314-8
    relatedItemType: JournalArticle
    title: Edible mycelium bioengineered for enhanced nutritional value and sensory
      appeal using a modular synthetic biology toolkit
  repository: PRIDE
  species:
  - NCBITaxID: 5062
    scientificName: Aspergillus oryzae
  title: Edible mycelium bioengineered for enhanced nutritional value and sensory
    appeal using a new synthetic biology toolkit
- bibliographicCitation: https://dx.doi.org/10.1016/J.CELREP.2023.113087
  brc: JBEI
  creator:
  - affiliation: Joint BioEnergy Institute, Lawrence Berkeley National Laboratory
      Berkeley CA 94720
    creatorName: Christopher J. Petzold
    email: cjpetzold@lbl.gov
    primaryContact: true
  date: '2024-02-28'
  identifier: PXD040697
  keywords:
  - ALE
  - CRISPR
  - Growth coupling
  - bioproduction
  - emergent cell phenotypes
  - genome scale metabolic models
  - heterologous pathway
  - indigoidine
  - lignin
  - proteomics analysis
  - recombineering
  - strain engineering
  relatedItem:
    relatedItemIdentifier: https://www.sciencedirect.com/science/article/pii/S2211124723010987
    relatedItemType: JournalArticle
    title: Maximizing microbial bioproduction from sustainable carbon sources using
      iterative systems engineering
  repository: PRIDE
  species:
  - NCBITaxID: 160488
    scientificName: Pseudomonas putida KT2440
  title: Ensemble and iterative engineering for maximal bioconversion to the blue
    pigment, Indigoidine from non-canonical sustainable carbon sources
- bibliographicCitation: https://dx.doi.org/10.1016/J.YMBEN.2023.11.009
  brc: JBEI
  creator:
  - affiliation: Joint BioEnergy Institute, Lawrence Berkeley National Laboratory
      Berkeley CA 94720
    creatorName: Christopher J. Petzold
    email: cjpetzold@lbl.gov
    primaryContact: true
  date: '2024-02-28'
  identifier: PXD044193
  keywords:
  - Citral
  - Citronellol
  - Corynebacterium glutamicum
  - Eucalyptol
  - Geranic acid
  - Geraniol
  - Linalool
  - Monoterpene biosynthetic production
  relatedItem:
    relatedItemIdentifier: https://www.sciencedirect.com/science/article/pii/S1096717623001738
    relatedItemType: JournalArticle
    title: Development of Corynebacterium glutamicum as a monoterpene production platform
  repository: PRIDE
  species:
  - NCBITaxID: 1718
    scientificName: Corynebacterium glutamicum
  title: Development of Corynebacterium glutamicum as a monoterpene production platform
- bibliographicCitation: https://dx.doi.org/10.6019/PXD041554
  brc: JBEI
  creator:
  - affiliation: Joint BioEnergy Institute, Lawrence Berkeley National Laboratory
      Berkeley CA 94720
    creatorName: Christopher J. Petzold
    email: cjpetzold@lbl.gov
    primaryContact: true
  date: '2023-09-07'
  identifier: PXD041554
  keywords:
  - CRISPRi
  - Carbon catabolite repression
  - Crc
  - Lignocellulose
  - Pseudomonas putida
  - small RNA
  relatedItem:
    relatedItemIdentifier: https://journals.asm.org/doi/10.1128/aem.00852-23
    relatedItemType: JournalArticle
    title: Simultaneous carbon catabolite repression governs sugar and aromatic co-utilization
      in Pseudomonas putida M2
  repository: PRIDE
  species:
  - NCBITaxID: 160488
    scientificName: Pseudomonas putida
  title: Simultaneous carbon catabolite repression governs sugar and aromatic co-utilization
    in Pseudomonas putida M2
- bibliographicCitation: https://dx.doi.org/10.6019/PXD031549
  brc: JBEI
  creator:
  - affiliation: Joint BioEnergy Institute, Lawrence Berkeley National Laboratory
      Berkeley CA 94720
    creatorName: Christopher J. Petzold
    email: cjpetzold@lbl.gov
    primaryContact: true
  date: '2023-05-10'
  identifier: PXD031549
  keywords:
  - Pseudomonas, Soil isolates, and arabinose, pentose, xylose
  relatedItem:
    relatedItemIdentifier: https://www.biorxiv.org/content/10.1101/2022.10.05.510959v1.full
    relatedItemType: JournalArticle
    title: Revealing Pentose Catabolism in Pseudomonas putida
  repository: PRIDE
  species:
  - NCBITaxID: 160488
    scientificName: Pseudomonas putida
  title: Revealing Pentose Catabolism in Pseudomonas putida
- bibliographicCitation: ' https,://dx.doi.org/10.6019/PXD031402'
  brc: JBEI
  creator:
  - affiliation: Joint BioEnergy Institute, Lawrence Berkeley National Laboratory
      Berkeley CA 94720
    creatorName: Christopher J. Petzold
    email: cjpetzold@lbl.gov
    primaryContact: true
  date: '2022-09-19'
  identifier: PXD031402
  keywords:
  - Biofuels
  - POP- FAME
  - Rocket fuel
  - polycyclopropanated fatty acid methyl esters
  - polyketide synthase
  - sustainability
  relatedItem:
    relatedItemIdentifier: https://www.sciencedirect.com/science/article/pii/S2542435122002380
    relatedItemType: JournalArticle
    title: Biosynthesis of polycyclopropanated high energy biofuels
  repository: PRIDE
  species:
  - NCBITaxID: 562
    scientificName: Escherichia coli
  title: Biosynthesis of polycyclopropanated high energy biofuels
- bibliographicCitation: https://proteomecentral.proteomexchange.org/cgi/GetDataset?ID=PXD024976
  brc: JBEI
  creator:
  - affiliation: Joint BioEnergy Institute, Lawrence Berkeley National Laboratory
      Berkeley CA 94720
    creatorName: Christopher J. Petzold
    email: cjpetzold@lbl.gov
    primaryContact: true
  date: '2022-09-09'
  identifier: PXD024976
  keywords:
  - Metabolic engineering, Synthetic biology, Vinblastine
  relatedItem:
    relatedItemIdentifier: https://www.nature.com/articles/s41586-022-05157-3
    relatedItemType: JournalArticle
    title: A microbial supply chain for production of the anti-cancer drug vinblastine
  repository: PanoramaPublic
  species:
  - NCBITaxID: 4932
    scientificName: Saccharomyces cerevisiae (Bakers yeast)
  title: A microbial supply chain for production of anti-cancer drug vinblastine
- bibliographicCitation: https://dx.doi.org/10.6019/PXD020215
  brc: JBEI
  creator:
  - affiliation: Joint BioEnergy Institute, Lawrence Berkeley National Laboratory
      Berkeley CA 94720
    creatorName: Christopher J. Petzold
    email: cjpetzold@lbl.gov
    primaryContact: true
  date: '2021-09-09'
  identifier: PXD020215
  keywords:
  - Enzyme assays
  - GHs (glycosyl hydrolases)
  - Laccases
  - Lignin
  - Multiplex assay
  - NIMS
  relatedItem:
    relatedItemIdentifier: https://www.nature.com/articles/s41598-021-91181-8
    relatedItemType: JournalArticle
    title: A multiplexed nanostructure-initiator mass spectrometry (NIMS) assay for
      simultaneously detecting glycosyl hydrolase and lignin modifying enzyme activities
  repository: PRIDE
  species:
  - NCBITaxID: 192525
    scientificName: Agaricus bisporus var. bisporus
  title: Development of a multiplexed platform to study laccases and cellulase activities
    using nanostructure-initiator mass spectrometry (NIMS)
- brc: JBEI
  creator:
  - affiliation: Joint BioEnergy Institute, Lawrence Berkeley National Laboratory
      Berkeley CA 94720
    creatorName: Christopher J. Petzold
    email: cjpetzold@lbl.gov
    primaryContact: true
  date: '2020-10-24'
  identifier: PXD021570
  keywords:
  - Metabolic engineering
  - genome editing
  relatedItem:
    relatedItemIdentifier: https://www.nature.com/articles/s41467-020-19171-4
    relatedItemType: JournalArticle
    title: Genome-scale metabolic rewiring improves titers rates and yields of the
      non-native product indigoidine at scale
  repository: PanoramaPublic
  species:
  - NCBITaxID: 160488
    scientificName: Pseudomonas putida KT2440
  title: Genome-scale metabolic rewiring to improve titers, rates and yields of the
    non-native non-ribosomal peptide product indigoidine at scale
- bibliographicCitation: https://dx.doi.org/10.6019/PXD013623
  brc: JBEI
  creator:
  - affiliation: Joint BioEnergy Institute, Lawrence Berkeley National Laboratory
      Berkeley CA 94720
    creatorName: Christopher J. Petzold
    email: cjpetzold@lbl.gov
    primaryContact: true
  date: '2020-04-07'
  identifier: PXD013623
  keywords:
  - NADH dehydrogenase
  - Pseudomonas
  - electron transport chain
  - oxidative stress
  - respiratory activity
  relatedItem:
    relatedItemIdentifier: https://journals.asm.org/doi/10.1128/aem.03038-19
    relatedItemType: JournalArticle
    title: "Systems Analysis of NADH Dehydrogenase Mutants Reveals Flexibility and\
      \ Limits of Pseudomonas taiwanensis VLB120\u2019s Metabolism"
  repository: PRIDE
  species:
  - NCBITaxID: 470150
    scientificName: Pseudomonas taiwanensis
  title: The metabolic response of Pseudomonas taiwanensis to NADH dehydrogenase deficiency
- bibliographicCitation: https,://dx.doi.org/10.6019/PXD014285
  brc: JBEI
  creator:
  - affiliation: Joint BioEnergy Institute, Lawrence Berkeley National Laboratory
      Berkeley CA 94720
    creatorName: Christopher J. Petzold
    email: cjpetzold@lbl.gov
    primaryContact: true
  date: '2020-04-07'
  identifier: PXD014285
  keywords:
  - Lignin valorization, Pseudomonas putida
  relatedItem:
    relatedItemIdentifier: https://chemistry-europe.onlinelibrary.wiley.com/doi/10.1002/cssc.202000268
    relatedItemType: JournalArticle
    title: Response of Pseudomonas putida to Complex, Aromatic-Rich Fractions from
      Biomass
  repository: PRIDE
  species:
  - NCBITaxID: 160488
    scientificName: Pseudomonas putida KT2440
  title: The global response of Pseudomonas putida to a complex lignolysate
- bibliographicCitation: https://proteomecentral.proteomexchange.org/cgi/GetDataset?ID=PXD013388
  brc: JBEI
  creator:
  - affiliation: Joint BioEnergy Institute, Lawrence Berkeley National Laboratory
      Berkeley CA 94720
    creatorName: Christopher J. Petzold
    email: cjpetzold@lbl.gov
    primaryContact: true
  date: '2020-02-11'
  identifier: PXD013388
  keywords:
  - Metabolic engineering
  - S. cerevisiae
  - Targeted proteomics
  - polyketides
  relatedItem:
    relatedItemIdentifier: https://www.sciencedirect.com/science/article/pii/S2405805X20300041
    relatedItemType: JournalArticle
    title: Programmable polyketide biosynthesis platform for production of aromatic
      compounds in yeast
  repository: PanoramaPublic
  species:
  - NCBITaxID: 4932
    scientificName: Saccharomyces cerevisiae (Bakers yeast)
  title: Programmable polyketide biosynthesis platform for production of aromatic
    compounds in yeast
- bibliographicCitation: https://proteomecentral.proteomexchange.org/cgi/GetDataset?ID=PXD012697
  brc: JBEI
  creator:
  - affiliation: Joint BioEnergy Institute, Lawrence Berkeley National Laboratory
      Berkeley CA 94720
    creatorName: Christopher J. Petzold
    email: cjpetzold@lbl.gov
    primaryContact: true
  date: '2019-10-29'
  identifier: PXD012697
  keywords:
  - HDMAPP
  - HIPP
  - Juvenile Hormones
  - Mevalonate Pathway
  - Terpenes
  - homodimethylallyl pyrophosphate
  - homoisopentenyl pyrophosphate
  relatedItem:
    relatedItemIdentifier: https://pubs.acs.org/doi/10.1021/acssynbio.9b00230
    relatedItemType: JournalArticle
    title: Mevalonate Pathway Promiscuity Enables Noncanonical Terpene Production
  repository: PanoramaPublic
  species:
  - NCBITaxID: 562
    scientificName: Escherichia coli
  title: Exploring Non-Canonical Terpene Space With a Heterologous Lepidopteran Mevalonate
    Pathway
- bibliographicCitation: https://proteomecentral.proteomexchange.org/cgi/GetDataset?ID=PXD014182
  brc: JBEI
  creator:
  - affiliation: Joint BioEnergy Institute, Lawrence Berkeley National Laboratory
      Berkeley CA 94720
    creatorName: Christopher J. Petzold
    email: cjpetzold@lbl.gov
    primaryContact: true
  date: '2019-08-23'
  identifier: PXD014182
  keywords:
  - Automation
  - Bacteria
  - Biotechnology
  - Fungi
  - High-throughput
  - Microbes
  - Sample preparation
  relatedItem:
    relatedItemIdentifier: https://pubs.acs.org/doi/10.1021/acs.jproteome.9b00455
    relatedItemType: JournalArticle
    title: "Automated \u201CCells-To-Peptides\u201D Sample Preparation Workflow for\
      \ High-Throughput, Quantitative Proteomic Assays of Microbes"
  repository: PanoramaPublic
  species:
  - NCBITaxID: 562
    scientificName: Escherichia coli
  title: An automated cells-to-peptides sample preparation workflow for high-throughput,
    quantitative proteomic assays of microbes
- bibliographicCitation: https://dx.doi.org/10.6019/PXD012013
  brc: JBEI
  creator:
  - affiliation: Joint BioEnergy Institute, Lawrence Berkeley National Laboratory
      Berkeley CA 94720
    creatorName: Christopher J. Petzold
    email: cjpetzold@lbl.gov
    primaryContact: true
  date: '2019-06-27'
  identifier: PXD012013
  keywords:
  - amino acids
  - biomass hydrolysates
  - lignin-related aromatics
  - methyl ketones
  - protein
  relatedItem:
    relatedItemIdentifier: https://analyticalsciencejournals.onlinelibrary.wiley.com/doi/10.1002/bit.26995
    relatedItemType: JournalArticle
    title: Methyl ketone production by Pseudomonas putida is enhanced by plant-derived
      amino acids
  repository: PRIDE
  species:
  - NCBITaxID: 160488
    scientificName: Pseudomonas putida KT2440
  title: Methyl ketone production by Pseudomonas putida is enhanced by plant-derived
    amino acids
- bibliographicCitation: https://dx.doi.org/10.6019/PXD011212
  brc: JBEI
  creator:
  - affiliation: Joint BioEnergy Institute, Lawrence Berkeley National Laboratory
      Berkeley CA 94720
    creatorName: Christopher J. Petzold
    email: cjpetzold@lbl.gov
    primaryContact: true
  date: '2019-06-27'
  identifier: PXD011212
  keywords:
  - high-throughput
  - workflow
  relatedItem:
    relatedItemIdentifier: https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0211582
    relatedItemType: JournalArticle
    title: A rapid methods development workflow for high-throughput quantitative proteomic
      applications
  repository: PRIDE
  species:
  - NCBITaxID: 562
    scientificName: Escherichia coli
  title: A rapid methods development workflow for high-throughput quantitative proteomic
    applications
- bibliographicCitation: https://proteomecentral.proteomexchange.org/cgi/GetDataset?ID=PXD011350
  brc: JBEI
  creator:
  - affiliation: Joint BioEnergy Institute, Lawrence Berkeley National Laboratory
      Berkeley CA 94720
    creatorName: Christopher J. Petzold
    email: cjpetzold@lbl.gov
    primaryContact: true
  date: '2019-05-17'
  identifier: PXD011350
  keywords:
  - targeted proteomics
  relatedItem:
    relatedItemIdentifier: https://journals.asm.org/doi/full/10.1128/mbio.02577-18
    relatedItemType: JournalArticle
    title: Massively Parallel Fitness Profiling Reveals Multiple Novel Enzymes in
      Pseudomonas putida Lysine Metabolism
  repository: PanoramaPublic
  species:
  - NCBITaxID: 160488
    scientificName: Pseudomonas putida KT2440
  title: Massively parallel fitness profiling reveals multiple novel enzymes in Pseudomonas
    putida lysine metabolism
- bibliographicCitation: https://proteomecentral.proteomexchange.org/cgi/GetDataset?ID=PXD012312
  brc: JBEI
  creator:
  - affiliation: Joint BioEnergy Institute, Lawrence Berkeley National Laboratory
      Berkeley CA 94720
    creatorName: Christopher J. Petzold
    email: cjpetzold@lbl.gov
    primaryContact: true
  date: '2019-05-16'
  identifier: PXD012312
  keywords:
  - Machine Learning
  - Metabolic engineering
  - Metabolomics
  - Protoemics
  - Synthetic biology
  relatedItem:
    relatedItemIdentifier: https://pubs.acs.org/doi/10.1021/acssynbio.9b00020
    relatedItemType: JournalArticle
    title: "Lessons from Two Design\u2013Build\u2013Test\u2013Learn Cycles of Dodecanol\
      \ Production in Escherichia coli Aided by Machine Learning"
  repository: PanoramaPublic
  species:
  - NCBITaxID: 562
    scientificName: Escherichia coli
  title: Lessons from Two Design-Build-Test-Learn Cycles of Dodecanol Production in
    Escherichia coli Aided by Machine Learning
- bibliographicCitation: https://proteomecentral.proteomexchange.org/cgi/GetDataset?ID=PXD013389
  brc: JBEI
  creator:
  - affiliation: Joint BioEnergy Institute, Lawrence Berkeley National Laboratory
      Berkeley CA 94720
    creatorName: Christopher J. Petzold
    email: cjpetzold@lbl.gov
    primaryContact: true
  date: '2019-04-16'
  identifier: PXD013389
  keywords:
  - RaxX
  - RiPPs
  - Tyrosine sulfation
  - X. oryzae
  - XA21 immune receptor
  relatedItem:
    relatedItemIdentifier: https://www.pnas.org/doi/10.1073/pnas.1818275116
    relatedItemType: JournalArticle
    title: Biosynthesis and secretion of the microbial sulfated peptide RaxX and binding
      to the rice XA21 immune receptor
  repository: PanoramaPublic
  species:
  - NCBITaxID: 360094
    scientificName: Xanthomonas oryzae pv. oryzae
  title: Biosynthesis and secretion of the microbial sulfated peptide RaxX and binding
    to the rice XA21 immune receptor
- bibliographicCitation: https://proteomecentral.proteomexchange.org/cgi/GetDataset?ID=PXD012293
  brc: JBEI
  creator:
  - affiliation: Joint BioEnergy Institute, Lawrence Berkeley National Laboratory
      Berkeley CA 94720
    creatorName: Christopher J. Petzold
    email: cjpetzold@lbl.gov
    primaryContact: true
  date: '2019-04-03'
  identifier: PXD012293
  keywords:
  - C. glutamicum
  - isoprenoid pathway
  - targeted proteomics
  relatedItem:
    relatedItemIdentifier: https://biotechnologyforbiofuels.biomedcentral.com/articles/10.1186/s13068-019-1381-3
    relatedItemType: JournalArticle
    title: Engineering Corynebacterium glutamicum to produce the biogasoline isopentenol
      from plant biomass hydrolysates
  repository: PanoramaPublic
  species:
  - NCBITaxID: 196627
    scientificName: Corynebacterium glutamicum ATCC 13032
  title: Engineering Corynebacterium glutamicum to produce the biogasoline isopentenol
    from plant biomass hydrolysates

```
## DatasetCollection-GLBRC_examples_negative
### Input
```yaml
datasets:
- analysisType: Expression profiling
  bibliographicCitation: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE201229
  brc: GLBRC
  creator:
  - affiliation: Great Lakes Bioenergy Research Center, University of Wisconsin -
      Madison
    creatorName: Myers, Kevin
    email: kmyers2@wisc.edu
    primaryContact: true
  date: '2022-04-24'
  description: The fermentation of Zymomonas mobilis 2032 was conducted in 6% and
    9% glucan-loading AFEX-pretreated corn stover hydrolysates, as well as in 7% glucan-loading
    AFEX-pretreated switchgrass hydrolysate, and RNA samples were collected at both
    glucose and xylose growth stages. Here we present a comprehensive RNA-Seq data
    set at both growth stages.
  identifier: GSE201229
  keywords:
  - AFEX
  - hydrolysate
  - RNA-Seq
  - Zymomonas
  relatedItem:
    relatedItemIdentifier: https://doi.org/10.1128/mra.00564-22
    relatedItemType: JournalArticle
    title: Transcriptomic Data Sets for Zymomonas mobilis 2032 during Fermentation
      of Ammonia Fiber Expansion (AFEX)-Pretreated Corn Stover and Switchgrass Hydrolysates
  repository: GEO
  species:
  - NCBITaxID: 542
    scientificName: Zymomonas mobilis
  title: Multistage Transcriptomics Data Set of Zymomonas mobilis 2032 During Fermentation
    in 6% and 9% Glucan-Loading AFEX-Pretreated Corn Stover Hydrolysates and 7% Glucan-Loading
    AFEX-Pretreated Switchgrass Hydrolysate
- analysisType: expression profiling
  bibliographicCitation: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE174697
  brc: GLBRC
  creator:
  - affiliation: Great Lakes Bioenergy Research Center, University of Wisconsin-Madison
    creatorName: Linz, Alexandra
    email: amlinz16@gmail.com
    primaryContact: true
  date: '2021-05-22'
  description: Aromatic diketones are a major product of formic acid lignin depolymerization.
    Novosphingobium aromaticivorans can degrade these diketones, but the enzymes used
    in this process were unknown. We used RNA-Seq to identify aromatic dimer dehydrogenases
    as potential candidates for the initial reduction of the aromatic G-diketone,
    then verified this using in vitro enzyme assays.
  identifier: GSE174697
  keywords:
  - Lignin
  - aromatic metabolism
  - Novosphingobium
  - sphingomonads
  - ketone reduction
  - aromatic dehydrogenases
  relatedItem:
    relatedItemIdentifier: https://doi.org/10.1128/AEM.01742-21
    relatedItemType: JournalArticle
    title: Aromatic Dimer Dehydrogenases from Novosphingobium aromaticivorans Reduce
      Monoaromatic Diketones
  repository: GEO
  species:
  - NCBITaxID: 48935
    scientificName: Novosphingobium aromaticivorans
  title: Aromatic dimer dehydrogenases from Novosphingobium aromaticivorans reduce
    aromatic diketones
- analysisType: Targeted Locus (Loci)
  bibliographicCitation: https://www.ncbi.nlm.nih.gov/bioproject?term=PRJNA785605
  brc: GLBRC
  creator:
  - affiliation: Great Lakes Bioenergy Research Center, Michigan State University
    creatorName: Gian Nico Benucci
    email: benucci@msu.edu
    primaryContact: true
  date: '2022-04-06'
  description: We sequenced the ITS and 16S rDNA region to characterize the microbial
    communities of switchgrass (Panicum virgatum), grown in the greenhouse, and investigate
    the effect of soil origin and plant genotype on microbiome assembly.
  identifier: PRJNA785605
  keywords:
  - bioenergy crops
  - sustainability
  - 16S rRNA
  - ITS
  - random forest
  - rhizobiome
  - plant biomass
  relatedItem:
    relatedItemIdentifier: https://doi.org/10.1128/mbio.00079-22
    relatedItemType: JournalArticle
    title: Soil Origin and Plant Genotype Modulate Switchgrass Aboveground Productivity
      and Root Microbiome Assembly
  repository: NCBI BioProject
  species:
  - NCBITaxID: 1348798
    scientificName: terrestrial metagenome
  title: Microbial communities of root and soil of switchgrass (Panicum virgatum)
    Targeted loci

```
## DatasetCollection-CABBI_examples_negative
### Input
```yaml
datasets:
- analysisType: shotgun_proteomics
  bibliographicCitation: https://dx.doi.org/10.6019/PXD021661
  brc: CABBI
  creator:
  - affiliation: Lewis-Sigler Institute for Integrative Genomics, Princeton University
    creatorName: Wuhr, Martin
    email: wuhr@princeton.edu
    primaryContact: true
  - affiliation: Princeton University
    creatorName: Stadlmeier, Michael
    email: ms93@princeton.edu
  date: '2021-05-11'
  description: The ability of the TMTPro isobaric labeling reagents to form complementary
    ions for accurate multiplexed proteomics at the MS2 level was investigated. Human
    and yeast peptides were labeled in distinct ratios to analyze the effect of interference
    on the quantification accuracy. A method, TMTProC, was developed and optimized
    for accurate, sensitive MS2-level quantification of up to 8 conditions in one
    MS-run.
  identifier: PXD021661
  keywords:
  - complementary ion quantification
  - interference-free quantification
  - multiplexing
  - TMTpro
  relatedItem:
    relatedItemIdentifier: https://dx.doi.org/10.1021/ACS.JPROTEOME.0C00813
    relatedItemType: JournalArticle
    title: TMTpro Complementary Ion Quantification Increases Plexing and Sensitivity
      for Accurate Multiplexed Proteomics at the MS2 Level.
  repository: PRIDE
  species:
  - NCBITaxID: 9606
    scientificName: Homo sapiens
  - NCBITaxID: 4932
    scientificName: Saccharomyces cerevisiae
  title: TMTPro Complementary Ion Quantification Increase Plexing and Sensitivity
    for Accurate Multiplexed Proteomics at the MS2 level
- analysisType: ''
  bibliographicCitation: https://doi.org/doi:10.25345/C5WV53
  brc: CABBI
  creator:
  - affiliation: Princeton University
    creatorName: Rabinowitz, Joshua
    email: joshr@princeton.edu
    primaryContact: true
  date: '2021-10-30'
  description: 'This dataset is associated with this manuscript: Metabolite discovery
    through global annotation of untargeted metabolomics data. Li Chen, Wenyun Lu,
    Lin Wang, Xi Xing, Xin Teng, Xianfeng Zeng, Antonio D Muscarella, Yihui Shen,
    Alexis Cowan, Melanie R McReynolds, Brandon Kennedy, Ashley M Lato, Shawn R Campagna,
    Mona Singh, Joshua Rabinowitz. The dataset has six main folders: (1) Yeast-MS1
    (2) Yeast-labeling (3) Liver-MS1 (4) Liver-targeted-MS2 (separate folders for
    mzxml files and matched csv files used as the inclusion list to set up PRM targeted-MS2
    runs) (5) Mouse-labeling (6) Glucosyl_taurine-liver-vs-synthesis. Please refer
    to README-raw-data-NetID-20210512.doc for more details.'
  identifier: MSV000087434
  keywords:
  - NetID
  - liver
  - yeast
  - full-scan-MS1
  - PRM-targeted-MS2
  relatedItem:
    relatedItemIdentifier: https://doi.org/10.1038/s41592-021-01303-3
    relatedItemType: JournalArticle
    title: Metabolite discovery through global annotation of untargeted metabolomics
      data
  repository: MassIVE
  species:
  - NCBITaxID: 10090
    scientificName: Mus musculus
  - NCBITaxID: 4932
    scientificName: Saccharomyces cerevisiae
  title: GNPS Raw data files for NetID manuscript

```
## DatasetCollection-CBI_examples_negative
### Input
```yaml
datasets:
- analysisType: image_analysis
  bibliographicCitation: https://dx.doi.org/10.25983/2263365
  brc: CBI
  creator:
  - affiliation: Center for Bioenergy Innovation, Oregon State University
    creatorName: Nagle, Michael F
    email: Michael.Nagle@libd.org
    primaryContact: 'No'
  - affiliation: Oregon State University
    creatorName: Strauss, Steven
    email: Steve.Strauss@oregonstate.edu
    primaryContact: 'Yes'
  date: '2023-12-27'
  description: This image dataset consists of 11,791 images representing 1,219 genotypes
    of Populus trichocarpa undergoing in planta regeneration. Genotypes were imaged
    with a median of four weekly timepoints and a median of two replicates each. A
    representative and diverse subset of 249 images was annotated using the IDEAS
    annotation interface (ideas.eecs.oregonstate.edu) and these annotated images were
    used to train a deep semantic segmentation model (PSPNet), which was deployed
    for inference over the entire dataset. Annotated classes include specific stages
    of regeneration (callus and shoot) in addition to unregenerated plant material
    and background.
  identifier: '2263365'
  keywords:
  - planta regeneration
  - callus
  - GWAS
  relatedItem:
    relatedItemIdentifier: https://dx.doi.org/10.1093/g3journal/kae026
    relatedItemType: JournalArticle
    title: GWAS supported by computer vision identifies large numbers of candidate
      regulators of in planta regeneration in Populus trichocarpa
  repository: OSTI
  species:
  - NCBITaxID: 3694
    scientificName: Populus trichocarpa
  title: Expanded phenomics (imaged based) dataset for Populus Trichocarpa
- analysisType: Genomic - SNP calling
  bibliographicCitation: https://dx.doi.org/10.25983/2263365
  brc: CBI
  creator:
  - affiliation: Center for Bioenergy Innovation, Oregon State University
    creatorName: Nagle, Michael F
    email: Michael.Nagle@libd.org
    primaryContact: 'No'
  - affiliation: Oregon State University
    creatorName: Strauss, Steven
    email: Steve.Strauss@oregonstate.edu
    primaryContact: 'Yes'
  date: '2021-10-30'
  description: 'The SNP dataset consists of over 40 million single-nucleotide polymorphisms
    across 1,323 wild accessions of Populus trichocarpa. It was merged with statistics
    from the image dataset associated with this DOI for the purposes of downstream
    genetic association mapping in a genome-wide association study. '
  identifier: '226365'
  keywords:
  - planta regeneration
  - callus
  - GWAS
  relatedItem:
    relatedItemIdentifier: https://dx.doi.org/10.1093/g3journal/kae026
    relatedItemType: JournalArticle
    title: GWAS supported by computer vision identifies large numbers of candidate
      regulators of in planta regeneration in Populus trichocarpa
  repository: OSTI
  species:
  - NCBITaxID: 3694
    scientificName: Populus Trichocarpa
  - {}
  title: Expanded genetic variation (SNP) dataset for Populus Trichocarpa

```
